graylog2-web-interface/src/views/components/searchbar/SearchButton.jsx

```diff
@@ -1,24 +1,60 @@
+// @flow strict
 import React from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import type { StyledComponent } from 'styled-components';
 
 import { Button } from 'components/graylog';
 import { Spinner, Icon } from 'components/common';
+import type { ThemeInterface } from 'theme/index';
 
-const SearchButton = ({ running, disabled, glyph }) => (
-  <Button type="submit" bsStyle={running ? 'warning' : 'success'} disabled={disabled} className="pull-left search-button-execute">
-    {running ? <Spinner fixedWidth pulse text="" /> : <Icon name={glyph} /> }
-  </Button>
-);
+const DirtyButton: StyledComponent<{}, ThemeInterface, any> = styled(Button)`
```

Comment:
Since you're not using the Theme here, you can just pass `void` instead of `ThemeInterface`